### PR TITLE
Add `all` support for Target Leakage

### LINKED ISSUE #4107: Add support to `all` for TargetLeakageDataCheck
As a user, i'd like to be able to pass `all` to TargetLeakageDC since it's an argument supported by Woodwork.

### PR DESCRIPTION
**Pull request by bchen1116**
fix #4107

### CODE REVIEW

**Comment by machineFL** [comment]
No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

File: evalml/data_checks/target_leakage_data_check.py [hunk 1]
```diff
@@ -21,12 +21,12 @@ class TargetLeakageDataCheck(DataCheck):
 
     Args:
         pct_corr_threshold (float): The correlation threshold to be considered leakage. Defaults to 0.95.
-        method (string): The method to determine correlation. Use 'max' for the maximum correlation, or for specific correlation metrics, use their name (ie 'mutual_info' for mutual information, 'pearson' for Pearson correlation, etc).
+        method (string): The method to determine correlation. Use 'all' or 'max' for the maximum correlation, or for specific correlation metrics, use their name (ie 'mutual_info' for mutual information, 'pearson' for Pearson correlation, etc).
```
Comment (christopherbunn) [inline evalml/data_checks/target_leakage_data_check.py:24]: Can we have a test to make sure that the results of setting `all` and `max` are the same?

**Review by christopherbunn — DISMISSED** [review]
Requested additional test case, but once that is addressed LGTM

**Review by jeremyliweishih — APPROVED** [review]
LGTM

**Review by christopherbunn — APPROVED** [review]

**Pull request merged**